front/lib/api/assistant/global_agents.ts

```diff
@@ -1202,27 +1236,40 @@ function _getManagedDataSourceAgent(
     };
   }
 
+  const actions: AgentActionConfigurationType[] = [];
+  if (searchMCPServerView) {
+    actions.push({
+      id: -1,
+      sId: agentId + "-search-action",
+      type: "mcp_server_configuration",
+      name: DEFAULT_RETRIEVAL_ACTION_NAME,
+      description: `The user's ${connectorProvider} data source.`,
+      mcpServerViewId: searchMCPServerView.sId,
+      internalMCPServerId: searchMCPServerView.internalMCPServerId,
+      dataSources: filteredDataSourceViews.map((dsView) => ({
+        sId: makeSId("data_source_configuration", {
+          id: dsView.id, // Wrong: here it should be agentDataSourceConfiguration.id
+          workspaceId: owner.id,
+        }),
```

Comment:
Opening a discussion here about this: https://dust4ai.slack.com/archives/C050SM8NSPK/p1749635113038149